packages/openactive-integration-tests/test/features/core/amending-order-quote/implemented/c2-with-different-details-test.js

```diff
@@ -0,0 +1,121 @@
+  // The primary opportunity criteria to use for the primary OrderItem under test
+  testOpportunityCriteria: 'TestOpportunityBookable',
+  // The secondary opportunity criteria to use for multiple OrderItem tests
+  controlOpportunityCriteria: 'TestOpportunityBookable',
+  // This test uses 2 opportunities, A & B
+  numOpportunitiesUsedPerCriteria: 2,
+},
+(configuration, orderItemCriteria, featureIsImplemented, logger) => {
+  // Both runs share the same UUID, so that the 2nd run is an amendment to the same Order
+  const uuid = generateUuid();
+
+  /**
+   * Note: This generates an it() block. Therefore, this must be run within a describe() block.
+   *
+   * @param {RequestState} state
+   * @param {C1 | B} stage
+   * @param {() => import('chakram').ChakramResponse} responseAccessorFn function that gets the stage's response (e.g. `() => state.c1Response`)
+   */
+  const itFeedItemAndResponseItemShouldMatchIds = (state, stage, responseAccessorFn) => {
+    Common.itForOrderItem(orderItemCriteria, state, stage, () => responseAccessorFn().body,
+      'ID should match the one specified in the open data feed',
+      (feedOrderItem, responseOrderItem) => {
+        chai.expect(responseOrderItem).to.nested.include({
+          'orderedItem.@id': feedOrderItem.orderedItem['@id'],
+        });
+      });
+  };
+
+  /**
+   * Create a new state and flow helper (to represent a distinct batch of opportunities)
+   * and then fetch some opportunities
+   *
+   * Note: This generates jest blocks like `beforeAll()`, `it()`, etc. Therefore, this must be run within a `describe()` block
+   */
+  function getOpportunityFeedItemsWithNewState() {
+    // Each scenario uses a separate state and flowHelper because they fetch separate opportunities
+    const state = new RequestState(logger, {
+      uuid,
+    });
+    const flow = new FlowHelper(state);
+
+    beforeAll(async () => {
+      await state.fetchOpportunities(orderItemCriteria);
+      await chakram.wait();
+    });
+
+    describe('Get Opportunity Feed Items', () => {
+      (new GetMatch({
+        state, flow, logger, orderItemCriteria,
+      }))
+        .beforeSetup()
+        .successChecks()
+        .validationTests();
+    });
+
+    return {
+      state, flow,
+    };
+  }
+
+  // N.B.: The following two tests must be performed sequentially - with
+  // Second Attempt occurring after First Attempt.
+  describe('First Attempt - C1', () => {
+    const { state, flow } = getOpportunityFeedItemsWithNewState();
+
+    describe('C1', () => {
+      (new C1({
+        state, flow, logger,
+      }))
+        .beforeSetup()
+        .successChecks()
+        .validationTests();
+    });
+  });
+
+  /** Fetch some new opportunities, amend the existing order with a C2 request, and then complete it */
+  describe('Second Attempt - C2 -> B', () => {
+    const { state, flow } = getOpportunityFeedItemsWithNewState();
+
+    describe('C2', () => {
```

Comment:
Just checking (not sure if we can easily test this? Perhaps with request logging?) that the Second Attempt does not call C1 within the second flow, and is happy to jump straight to C2 and ignore C1?

We mentioned previously that C1 might be automatically called if it was not run within a particular flow?

Reply:
Good point, this will call C1. That's a separate test. I'll update this one

Reply:
With the latest change, I checked the test log and there was only one C1 request logged (https://openactive-reference-implementation.azurewebsites.net/api/openbooking/order-quote-template/{uuid})